pallets/parachain-staking/src/delegation_requests.rs

```diff
@@ -333,6 +333,13 @@ impl<T: Config> Pallet<T> {
 		let now = <Round<T>>::get().current;
 		let when = now.saturating_add(T::LeaveDelegatorsDelay::get());
 
+		// lazy migration for DelegatorStatus::Leaving
+		#[allow(deprecated)]
+		if matches!(state.status, DelegatorStatus::Leaving(_)) {
+			state.status = DelegatorStatus::Active;
+			<DelegatorState<T>>::insert(delegator.clone(), state.clone());
```

Comment:
@crystalin This kind of the same behavior - during schedule we preemptively overwrite the status to active. There are very few unlikely reasons for this extrinsic to fail (`DelegationDNE`, or `DelegatorAlreadyLeaving` if they manually scheduled all revokes) but if it does the user will get an error but their status would already be updated.

Reply:
In the case of `DelegatorAlreadyLeaving` it is fine because it means the user will be able to execute leave through the new flow.
In the case of `DelegationDNE` it should not be possible as in RT150x you can only leave if you have a delegation no ?

Reply:
Not quite, in RT150x you can execute leave as long as you are a delegator. But on a closer look `DelegationDNE` can never happen since the code reads first from `state.delegations` then checks again in `state.delegation` (it can be simplified, but probably in a different PR?)

```rust
        // state
        pub fn get_bond_amount(&self, collator: &AccountId) -> Option<Balance> {
		self.delegations          // <-state.delegations
			.0
			.iter()
			.find(|b| &b.owner == collator)
			.map(|b| b.amount)
	}

        for bond in state.delegations.0.clone() {   // <- state.delegations
		let collator = bond.owner;
		let mut scheduled_requests = <DelegationScheduledRequests<T>>::get(&collator);
		let bonded_amount = state
			.get_bond_amount(&collator)
			.ok_or(<Error<T>>::DelegationDNE)?; // <- state.delegations, this could be simplified to bond.amount
        }
```